lib/ae_mdw/db/util.ex

```diff
@@ -107,6 +112,55 @@ defmodule AeMdw.Db.Util do
     end
   end
 
+  @spec key_block_height(state(), binary()) :: {:ok, Blocks.height()} | {:error, Error.t()}
+  def key_block_height(state, hash_or_kbi) do
+    case Util.parse_int(hash_or_kbi) do
+      {:ok, kbi} when kbi >= 0 ->
+        last_gen = last_gen(state)
+
+        if kbi <= last_gen do
+          {:ok, kbi}
+        else
+          {:error, ErrInput.NotFound.exception(value: hash_or_kbi)}
+        end
+
+      {:ok, _kbi} ->
+        {:error, ErrInput.NotFound.exception(value: hash_or_kbi)}
+
+      :error ->
+        with {:ok, height, _hash} <- extract_height_hash(state, :key, hash_or_kbi) do
+          {:ok, height}
+        end
+    end
+  end
+
+  @spec micro_block_height_index(state(), binary()) ::
+          {:ok, Blocks.height(), Blocks.mbi()} | {:error, Error.t()}
+  def micro_block_height_index(state, mb_hash) do
+    with {:ok, height, decoded_hash} <- extract_height_hash(state, :micro, mb_hash) do
+      mbi =
+        decoded_hash
+        |> Db.get_micro_blocks()
+        |> Enum.count()
```

Comment:
Same reasoning as https://github.com/aeternity/ae_mdw/pull/898#discussion_r969051347

The proposed change is to add this to `AeMdw.Node.Db`:
``` 
  def micro_block_index(hash), do: micro_block_counter(hash, -1)

  defp micro_block_counter(hash, acc) do
    with block <- :aec_db.get_block(hash),
         :micro <- :aec_blocks.type(block) do
      micro_block_counter(:aec_blocks.prev_hash(block), acc + 1)
    else
      :key -> acc
    end
  end
  ```
and use `micro_block_index(hash)` instead of calling `Db.get_micro_blocks() |> Enum.count()`

Similar to:
```
defp micro_block_walker(hash) do
    with block <- :aec_db.get_block(hash),
         :micro <- :aec_blocks.type(block) do
      {block, :aec_blocks.prev_hash(block)}
    else
      :key -> nil
    end
  end
  ```
  but without accumulating all previous micro blocks and transactions.

Reply:
Discussed in previous PR